File: tip_calculator.py
# tip_calculator.py
import sys

def get_float(prompt):
    while True:
        try:
            return float(input(prompt))
        except ValueError:
            print("Invalid input. Please enter a number.")

def get_int(prompt):
    while True:
        try:
            val = int(input(prompt))
            if val <= 0:
                print("Must be positive.")
                continue
            return val
        except ValueError:
            print("Invalid input. Please enter an integer.")

def main():
    print("=== Tip Calculator ===")
    bill = get_float("Enter bill amount: ")
    tip_percent = get_float("Enter tip percentage (e.g., 15 for 15%): ")
    people = get_int("Enter number of people: ")

    tip = bill * (tip_percent / 100)
    total = bill + tip
    tip_per_person = tip / people
    total_per_person = total / people

    print("\n--- Result ---")
    print(f"Bill: ${bill:.2f}")
    print(f"Tip ({tip_percent:.0f}%): ${tip:.2f}")
    print(f"Total: ${total:.2f}")
    print(f"Split among {people} people:")
    print(f"Tip per person: ${tip_per_person:.2f}")
    print(f"Total per person: ${total_per_person:.2f}")

if __name__ == "__main__":
    main()

File: TipCalculator.cs
// TipCalculator.cs
using System;

class TipCalculator
{
    static double GetDouble(string prompt)
    {
        while (true)
        {
            Console.Write(prompt);
            string input = Console.ReadLine();
            if (double.TryParse(input, out double result) && result >= 0)
                return result;
            Console.WriteLine("Invalid input. Please enter a positive number.");
        }
    }

    static int GetInt(string prompt)
    {
        while (true)
        {
            Console.Write(prompt);
            string input = Console.ReadLine();
            if (int.TryParse(input, out int result) && result > 0)
                return result;
            Console.WriteLine("Invalid input. Please enter a positive integer.");
        }
    }

    static void Main()
    {
        Console.WriteLine("=== Tip Calculator ===");
        double bill = GetDouble("Enter bill amount: ");
        double tipPercent = GetDouble("Enter tip percentage (e.g., 15 for 15%): ");
        int people = GetInt("Enter number of people: ");

        double tip = bill * (tipPercent / 100);
        double total = bill + tip;
        double tipPerPerson = tip / people;
        double totalPerPerson = total / people;

        Console.WriteLine("\n--- Result ---");
        Console.WriteLine($"Bill: ${bill:F2}");
        Console.WriteLine($"Tip ({tipPercent:F0}%): ${tip:F2}");
        Console.WriteLine($"Total: ${total:F2}");
        Console.WriteLine($"Split among {people} people:");
        Console.WriteLine($"Tip per person: ${tipPerPerson:F2}");
        Console.WriteLine($"Total per person: ${totalPerPerson:F2}");
    }
}
